provider/openrouter/openrouter.go

```diff
@@ -51,14 +54,37 @@ func (p *Provider) ID() (id string) {
 	return "openrouter"
 }
 
+// ModelsList holds a list of models.
+type ModelsList struct {
+	Models []Model `json:"data"`
+}
+
+// Model holds a model.
+type Model struct {
+	// ID holds the model id.
+	ID string `json:"id"`
+	// Pricing holds the pricing information of a model.
+	Pricing Pricing `json:"pricing"`
+}
+
+// Pricing holds the pricing information of a model.
+type Pricing struct {
+	// Prompt holds the price for a prompt.
+	Prompt string `json:"prompt"`
+	// Completion holds the price for a completion.
+	Completion string `json:"completion"`
+	// Request holds the price for a request.
+	Request string `json:"request"`
+	// Image holds the price for a image.
+	Image string `json:"image"`
+}
+
 // Models returns which models are available to be queried via this provider.
 func (p *Provider) Models() (models []model.Model, err error) {
-	client := p.client()
-
-	var responseModels openai.ModelsList
+	var responseModels ModelsList
 	if err := retry.Do( // Query available models with a retry logic cause "openrouter.ai" has failed us in the past.
 		func() error {
-			ms, err := client.ListModels(context.Background())
+			ms, err := providerModels(p.baseURL + "/models")
```

Comment:
Commit message nitpick: would not use the word "new" cause usually everything in a commit is new.

How about `Custom HTTP request to list openrouter models, so pricing` ?